Assets/Scripts/Effects/SpawnParticleEffect.cs

```diff
@@ -1,50 +1,44 @@
-﻿using System;
+﻿using MapTools;
 using UnityEngine;
 
 namespace Effects
 {
     public class SpawnParticleEffect : MonoBehaviour
     {
-        [SerializeField]
-        private GameObject spawnParticleEffect;
-        [SerializeField]
-        private bool onDestroy;
-        [SerializeField]
-        private bool onCollision;
-        [SerializeField]
-        private bool useTag;
-        [SerializeField]
-        private new string tag = string.Empty;
-        
-        
+        [SerializeField] private GameObject spawnParticleEffect;
+        [SerializeField] private bool onDestroy;
+        [SerializeField] private bool onCollision;
+        [SerializeField] private bool useTag;
+        [SerializeField] private new string tag = string.Empty;
+
         private void OnDestroy()
         {
             if (onDestroy)
-            { 
-                SpawnEffect();  
+            {
+                SpawnEffect();
             }
         }
 
         private void OnTriggerEnter2D(Collider2D col)
         {
             if (!onDestroy && onCollision && !useTag)
-            { 
-                SpawnEffect();  
+            {
+                SpawnEffect();
             }
-            if (!onDestroy && onCollision && useTag)
+
+            if (onDestroy || !onCollision || !useTag) return;
+            if (col.CompareTag(tag))
```

Comment:
useTag && col.CompareTag or comment that is is implied that useTag is true

Reply:
reworked

Reply:
looks nice